packages/lib/src/vis/shared/PanZoomMesh.tsx

```diff
@@ -43,16 +48,7 @@ function PanZoomMesh() {
       camera.updateMatrixWorld();
       invalidate();
     },
-    [
-      abscissaScale,
-      camera,
-      height,
-      invalidate,
-      ordinateScale,
-      visWidth,
-      visHeight,
-      width,
-    ]
+    [abscissaScale, ordinateScale, camera, visWidth, visHeight, invalidate]
```

Comment:
Maybe the destructuring of `position` and `projectionMatrixInverse` should be done outside `useCallback` so the two objects can be referenced as dependencies (the `camera` object probably gets mutated by Three).

Reply:
Is it really needed given that `camera` is already needed as dependencies due to `camera.updateMatrixWorld()` below ?

Reply:
Totally, you're right. `camera.*` don't re-render the component, so there's no point in having them as dependencies (since the `useCallback` won't ever know they have changed). https://epicreact.dev/why-you-shouldnt-put-refs-in-a-dependency-array/

Reply:
This is why ESLint doesn't complain about `viewportCenter` not being a dep (it knows it's a mutable ref). Live and learn.